Scripts/RemediationScripts/Remediate-SetServiceBusMinReqTLSVersion.ps1

```diff
@@ -0,0 +1,813 @@
+    Write-Host $([Constants]::SingleDashLine)	
+
+    $colsProperty = @{Expression = { $_.ResourceName }; Label = "ResourceName"; Width = 30; Alignment = "left" },
+    @{Expression = { $_.ResourceGroupName }; Label = "ResourceGroupName"; Width = 30; Alignment = "left" },
+    @{Expression = { $_.ResourceId }; Label = "ResourceId"; Width = 100; Alignment = "left" },
+    @{Expression = { $_.MinimumTLSVersion }; Label = "MinimumTLSVersion"; Width = 100; Alignment = "left" }
+
+    if (-not $AutoRemediation) {
+        Write-Host "Azure Service Bus Namespace(s) with non-secure TLS version enabled are:"
+        $actionableServiceBusNamespacesDetails  | Format-Table -Property $colsProperty -Wrap
+        Write-Host $([Constants]::SingleDashLine)
+    }
+
+    # Back up snapshots to `%LocalApplicationData%'.
+    $backupFolderPath = "$([Environment]::GetFolderPath('LocalApplicationData'))\AzTS\Remediation\Subscriptions\$($context.Subscription.SubscriptionId.replace('-','_'))\$($(Get-Date).ToString('yyyyMMddhhmm'))\Set-SecureTLSVersionForServiceBusNamespaces"
+
+    if (-not (Test-Path -Path $backupFolderPath)) {
+        New-Item -ItemType Directory -Path $backupFolderPath | Out-Null
+    }
+
+    Write-Host "[Step 3 of 4] Back up Azure Service Bus Namespace(s) details"
+    Write-Host $([Constants]::SingleDashLine)
+
+    if ([String]::IsNullOrWhiteSpace($InputFilePath)) {
+        # Backing up actionable Azure Service Bus Namespace(s) details.
+        $backupFile = "$($backupFolderPath)\ServiceBusNamespacesDetailsBackUp.csv"
+        $actionableServiceBusNamespacesDetails | Export-CSV -Path $backupFile -NoTypeInformation
+        Write-Host "Azure Service Bus Namespace(s) details have been backed up to [$($backupFile)]" -ForegroundColor $([Constants]::MessageType.Update)
+        Write-Host $([Constants]::SingleDashLine)
+    }
+    else {
+        Write-Host "Skipped as -InputFilePath is provided" -ForegroundColor $([Constants]::MessageType.Warning)
+        Write-Host $([Constants]::SingleDashLine)
+    }
+
+    if (-not $DryRun) {
+        Write-Host $([Constants]::DoubleDashLine)
+        Write-Host "[Step 4 of 4] Enable secure TLS version on Azure Service Bus Namespace(s) in the Subscription" 
+        Write-Host $([Constants]::SingleDashLine)
+
+
+        if (-not $Force) {
+            Write-Host "Do you want to enable secure TLS version on Azure Service Bus Namespace(s) in the Subscription? " -ForegroundColor $([Constants]::MessageType.Warning)
+
+            $userInput = Read-Host -Prompt "(Y|N)"
+
+            if ($userInput -ne "Y") {
+                Write-Host "TLS version will not be changed for Azure Service Bus Namespace(s) in the Subscription. Exiting..." -ForegroundColor $([Constants]::MessageType.Warning)
+                Write-Host $([Constants]::DoubleDashLine)	
+                return
+            }
+        }
+        else {
+            if(-not $AutoRemediation){
```

Comment:
This is also correct but in other BRS we put this condition above such that it encloses if..else(-not Force) block.